tools/image-tool/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+use std::{
+    env,
+    path::{Path, PathBuf},
+    rc::Rc,
+};
+
+use async_trait::async_trait;
+use crane::CraneCLI;
+use docker::DockerCLI;
+use snafu::{ensure, ResultExt};
+use tokio::process::Command;
+use which::which;
+
+mod crane;
+mod docker;
+
+#[async_trait]
+pub trait ImageToolImpl {
+    /// Get the path of the cli tool to call
+    fn path(&self) -> PathBuf;
+
+    /// Pull an image archive to disk
+    async fn pull_archive(&self, file: &Path, uri: &str, cache: &Path) -> Result<()>;
+    /// Fetch the manifest
+    async fn get_manifest(&self, uri: &str) -> Result<Vec<u8>>;
+    /// Fetch the image config
+    async fn get_config(&self, uri: &str) -> Result<Vec<u8>>;
+
+    async fn output(&self, args: &[&str], error_msg: String) -> Result<Vec<u8>> {
```

Comment:
I wonder if we should avoid exposing this in the trait? Seems like we want callers to use one of the abstract methods, and that this `output` function is more of an implementation detail for the impls we have. (same with `spawn`)